src/Front/Pattern/Desugar.hs

```diff
@@ -79,11 +79,27 @@ desugarTopDefs = mapM desugarTopDef
 desugarDefs :: [A.Definition] -> DesugarFn [A.Definition]
 desugarDefs = mapM desugarDef
 
--- WARN: only body is desugared
 desugarDef :: A.Definition -> DesugarFn A.Definition
-desugarDef (A.DefFn i ps t e) = A.DefFn i ps t <$> desugarExpr e
+desugarDef (A.DefFn i [] t e) = return (A.DefFn i [] t e)
+desugarDef (A.DefFn i ps t e) =
+  let (pats, rese) = helper ps e (0::Int) in
+    A.DefFn i pats t <$> desugarExpr rese
+  where helper [] ex _ = ([],ex)
```

Comment:
Are you able to get rid of this case since `ps` is never empty, or do you get  warning?